pkg/pillar/zedbox/zedbox.go

```diff
@@ -32,45 +37,155 @@ import (
 	"github.com/lf-edge/eve/pkg/pillar/cmd/zedagent"
 	"github.com/lf-edge/eve/pkg/pillar/cmd/zedmanager"
 	"github.com/lf-edge/eve/pkg/pillar/cmd/zedrouter"
+	"github.com/lf-edge/eve/pkg/pillar/pidfile"
 	"github.com/lf-edge/eve/pkg/pillar/pubsub"
 	"github.com/lf-edge/eve/pkg/pillar/pubsub/socketdriver"
+	log "github.com/sirupsen/logrus"
 )
 
-var entrypoints = map[string]func(*pubsub.PubSub){
-	"client":           client.Run,
-	"command":          command.Run,
-	"diag":             diag.Run,
-	"domainmgr":        domainmgr.Run,
-	"downloader":       downloader.Run,
-	"executor":         executor.Run,
-	"hardwaremodel":    hardwaremodel.Run,
-	"ledmanager":       ledmanager.Run,
-	"logmanager":       logmanager.Run,
-	"nim":              nim.Run,
-	"nodeagent":        nodeagent.Run,
-	"verifier":         verifier.Run,
-	"volumemgr":        volumemgr.Run,
-	"waitforaddr":      waitforaddr.Run,
-	"zedagent":         zedagent.Run,
-	"zedmanager":       zedmanager.Run,
-	"zedrouter":        zedrouter.Run,
-	"ipcmonitor":       ipcmonitor.Run,
-	"baseosmgr":        baseosmgr.Run,
-	"wstunnelclient":   wstunnelclient.Run,
-	"conntrack":        conntrack.Run,
-	"tpmmgr":           tpmmgr.Run,
-	"vaultmgr":         vaultmgr.Run,
-	"upgradeconverter": upgradeconverter.Run,
-}
+const (
+	agentName   = "zedbox"
+	socketPath  = "/var/run"
+	errorTime   = 3 * time.Minute
+	warningTime = 40 * time.Second
+)
+
+var (
+	debugOverride = false
+	//Version Set from Makefile
+	Version     = "No version specified"
+	entrypoints = map[string]func(*pubsub.PubSub){
+		"client":           client.Run,
+		"command":          command.Run,
+		"diag":             diag.Run,
+		"domainmgr":        domainmgr.Run,
+		"downloader":       downloader.Run,
+		"executor":         executor.Run,
+		"hardwaremodel":    hardwaremodel.Run,
+		"ledmanager":       ledmanager.Run,
+		"logmanager":       logmanager.Run,
+		"nim":              nim.Run,
+		"nodeagent":        nodeagent.Run,
+		"verifier":         verifier.Run,
+		"volumemgr":        volumemgr.Run,
+		"waitforaddr":      waitforaddr.Run,
+		"zedagent":         zedagent.Run,
+		"zedmanager":       zedmanager.Run,
+		"zedrouter":        zedrouter.Run,
+		"ipcmonitor":       ipcmonitor.Run,
+		"baseosmgr":        baseosmgr.Run,
+		"wstunnelclient":   wstunnelclient.Run,
+		"conntrack":        conntrack.Run,
+		"tpmmgr":           tpmmgr.Run,
+		"vaultmgr":         vaultmgr.Run,
+		"upgradeconverter": upgradeconverter.Run,
+	}
+)
 
 func main() {
+	log.Infof("Starting %s", agentName)
```

Comment:
Again, can we retain (via non-default cmd flag) a capability of having old stype zebox behavior?

Reply:
Or maybe even -- lets call new binary pillar and the old school binary will still be called zedbox. That way, you can have your cake and eat it too (build zedbox as it used to be and have pillar -> zedbox)

Reply:
@rvs Once this is in master we can do a config file or something which can specify how many binaries to run and the mapping from microservices to zedbox processes. Shouldn't be too hard given the infrastructure in this PR